### PR TITLE
Issue-601 New RPC endpoints

### PR DESCRIPTION
**Pull request by kjaiswal**
Thank you for your contribution to Taquito.

Before submitting this PR, please make sure:

- [ ] Your code builds cleanly without any errors or warnings
- [ ] You have added unit tests
- [ ] You have added integration tests (if relevant/appropriate)
- [ ] All public methods or types have TypeDoc coverage with a complete description, and ideally an @example
- [ ] You have added or updated corresponding documentation
- [ ] If relevant, you have written a first draft summary describing the change for inclusion in Release Notes. 

## Release Note Draft Snippet

__If relevant, please write a summary of your change that will be suitable for
inclusion in the Release Notes for the next Taquito release.__

### CODE REVIEW

File: packages/taquito-rpc/src/taquito-rpc.ts [hunk 1]
```diff
@@ -779,4 +780,41 @@ export class RpcClient {
     return this.url
   }
 
+  /**
+   *
+   * @param options contains generic configuration for rpc calls
+   *
+   * @description Voting period of current block.
+   *
+   * @see https://tezos.gitlab.io/008/rpc.html#get-block-id-votes-current-period
```
Comment (jevonearth) [inline packages/taquito-rpc/src/taquito-rpc.ts:789]: @kjaiswal 

Can you update the URLs to use `api` in place of the proto number? ie `https://tezos.gitlab.io/api/rpc.html#get-block-id-votes-current-period` 

The API website should always redirect the API to the latest protocol, so it should mean less maintenance for us.

Comment (jevonearth) [reply packages/taquito-rpc/src/taquito-rpc.ts:789]: Can you add `@example` props too?

Adding an `@example` prop for the typical case (head) and for a specific block is very helpful to developers.

**Review by jevonearth — CHANGES_REQUESTED** [review]
Looks great, two minor requests Re Typedocs

File: packages/taquito-rpc/src/types.ts [hunk 2]
```diff
@@ -677,7 +687,9 @@ export interface BlockMetadata {
   max_operation_list_length: MaxOperationListLength[];
   baker: string;
   level: Level;
+  level_info: LevelInfo;
```
Comment (roxaneletourneau) [inline packages/taquito-rpc/src/types.ts:690]: I think level, level_info, voting_period_kind, and voting_period_info should be optional to stay compatible.


Deprecated: metadata.level, metadata.voting_period_kind
New properties: metadata.level_info, metadata.voting_period_info

File: packages/taquito-rpc/src/taquito-rpc.ts (hunk 1 above)
Comment (kjaiswal) [reply packages/taquito-rpc/src/taquito-rpc.ts]: Thank you for the feedback I have added @example props and replaced 008 with api.

File: packages/taquito-rpc/src/types.ts (hunk 2 above)
Comment (kjaiswal) [reply packages/taquito-rpc/src/types.ts]: Thank you for the feedback, I have made these properties optional.

**Review by jevonearth — APPROVED** [review]

**Pull request merged**